File: backend/src/models/DepartmentMember.js
'use strict';

const { Model, DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  class DepartmentMember extends Model {
    static associate(models) {
      // DepartmentMember belongs to Member
      DepartmentMember.belongsTo(models.Member, {
        foreignKey: 'member_id',
        as: 'member'
      });

      // DepartmentMember belongs to Department
      DepartmentMember.belongsTo(models.Department, {
        foreignKey: 'department_id',
        as: 'department'
      });
    }
  }

  DepartmentMember.init({
    id: {
      type: DataTypes.BIGINT,
      primaryKey: true,
      autoIncrement: true,
      allowNull: false
    },
    department_id: {
      type: DataTypes.BIGINT,
      allowNull: false,
      field: 'department_id'
    },
    member_id: {
      type: DataTypes.BIGINT,
      allowNull: false,
      field: 'member_id'
    },
    role_in_department: {
      type: DataTypes.STRING(50),
      allowNull: false,
      defaultValue: 'member',
      field: 'role_in_department'
    },
    joined_at: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
      field: 'joined_at'
    },
    status: {
      type: DataTypes.STRING(20),
      allowNull: false,
      defaultValue: 'active',
      validate: {
        isIn: [['active', 'inactive', 'pending']]
      }
    },
    notes: {
      type: DataTypes.TEXT,
      allowNull: true
    }
  }, {
    sequelize,
    modelName: 'DepartmentMember',
    tableName: 'department_members',
    timestamps: true,
    underscored: true
  });

  return DepartmentMember;
};

File: backendJava/src/main/java/church/abunearegawi/backend/model/DepartmentMember.java
package church.abunearegawi.backend.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "department_members")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DepartmentMember {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @Column(name = "role_in_department", nullable = false, length = 50)
    @Builder.Default
    private String roleInDepartment = "member";

    @Column(name = "joined_at", nullable = false)
    @Builder.Default
    private LocalDateTime joinedAt = LocalDateTime.now();

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private Status status = Status.active;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum Status {
        active, inactive, pending
    }
}
